### PR TITLE
fix disk fault recovery

### PR DESCRIPTION
**Pull request by panda-sheep**
## What type of PR is this?
- [ ] bug
- [x] feature
- [ ] enhancement

#### Description:
fix  https://github.com/vesoft-inc/nebula/issues/4123

When a storage configure contains multiple dataPaths, this pr handles adding dataPaths or removing some dataPaths.

For example: 
dataPath in storage configure contains 2 dataPaths, namely dataPath1 and dataPath2, and each dataPath contains 50 parts.
At this time, dataPath3 was newly added.
                     
case1: The disk of dataPath1 is broken, but there is no repair. 
            Remove dataPath1 from dataPath in storage configure.
case2: The disk of dataPath1 is broken, but repaired.

At this time, the storage is started, and the distribution of the final part is as follows:
image

## Checklist:
Tests:
- [ ] Unit test(positive and negative cases)
- [ ] Function test
- [ ] Performance test
- [ ] N/A

Affects:
- [ ] Documentation affected (Please add the label if documentation needs to be modified.)
- [ ] Incompatibility (If it breaks the compatibility, please describe it and add the label.）
- [ ] If it's needed to cherry-pick (If cherry-pick to some branches is required, please label the destination version(s).)
- [ ] Performance impacted: Consumes more CPU/Memory

## Release notes:

Please confirm whether to be reflected in release notes and how to describe:
> ex. Fixed the bug .....

### CODE REVIEW

**Comment by panda-sheep** [comment]
Has tested, ready to review

File: src/kvstore/NebulaStore.cpp [hunk 1]
```diff
@@ -359,14 +359,31 @@ ErrorOr<nebula::cpp2::ErrorCode, HostAddr> NebulaStore::partLeader(GraphSpaceID
...
+    // Iterate over all engines to ensure that each dataPath has an engine
     if (this->spaces_.find(spaceId) != this->spaces_.end()) {
       LOG(INFO) << "Data space " << spaceId << " has existed!";
-      return;
-    }
-    LOG(INFO) << "Create data space " << spaceId;
-    this->spaces_[spaceId] = std::make_unique<SpacePartInfo>();
-    for (auto& path : options_.dataPaths_) {
-      this->spaces_[spaceId]->engines_.emplace_back(newEngine(spaceId, path, options_.walPath_));
+      for (auto& path : options_.dataPaths_) {
```
Comment (critical27) [inline src/kvstore/NebulaStore.cpp:365]: Why not build the engine in `loadPartFromPartManager`, `addSpace` is called for each partition in meta, which is duplicated.

Comment (critical27) [reply src/kvstore/NebulaStore.cpp:365]: Would you add a test case if possible.

Comment (panda-sheep) [reply src/kvstore/NebulaStore.cpp:365]: First, `addSpace` is processed once per spaceId
Secondly, the `addSpace` interface is to create an engine for each dataPath of the current host, check and create an engine according to the dataPath and put it in addSpace to feel more unified.
wdyt?

Comment (critical27) [reply src/kvstore/NebulaStore.cpp:365]: I got it, my bad.

File: src/kvstore/NebulaStore.cpp [hunk 2]
```diff
@@ -359,14 +364,31 @@ ErrorOr<nebula::cpp2::ErrorCode, HostAddr> NebulaStore::partLeader(GraphSpaceID
 void NebulaStore::addSpace(GraphSpaceID spaceId, bool isListener) {
   folly::RWSpinLock::WriteHolder wh(&lock_);
   if (!isListener) {
+    // Iterate over all engines to ensure that each dataPath has an engine
```
Comment (wenhaocs) [inline src/kvstore/NebulaStore.cpp:367]: It's better to make it clear in the comments that disk paths may not have kv engines created after recovering from repair.

**Review by wenhaocs — APPROVED** [review]
LGTM. Per discussion, it's better to make it clear that disks are empty after repair.

**Review by critical27 — APPROVED** [review]
Good job~

**Pull request merged**